File: app/src/main/java/com/tt/handsomeman/adapter/CategoryAdapter.java
package com.tt.handsomeman.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.tt.handsomeman.databinding.ItemCategoryBinding;
import com.tt.handsomeman.model.Category;

import java.util.List;

public class CategoryAdapter extends RecyclerView.Adapter<CategoryAdapter.CategoryViewHolder> {

    private List<Category> categoryList;
    private LayoutInflater layoutInflater;
    private Context context;
    private ItemCategoryBinding binding;

    private OnItemClickListener mListener;

    public CategoryAdapter(Context context,
                           List<Category> categoryList) {
        this.categoryList = categoryList;
        this.context = context;
        layoutInflater = LayoutInflater.from(context);
    }

    public void setOnItemClickListener(OnItemClickListener listener) {
        mListener = listener;
    }

    @NonNull
    @Override
    public CategoryViewHolder onCreateViewHolder(@NonNull ViewGroup parent,
                                                 int viewType) {
        binding = ItemCategoryBinding.inflate(layoutInflater, parent, false);
        View item = binding.getRoot();
        return new CategoryViewHolder(item, mListener);
    }

    @Override
    public void onBindViewHolder(@NonNull CategoryViewHolder holder,
                                 int position) {
        Category category = categoryList.get(position);
        holder.tvCategoryName.setText(category.getName());
    }

    @Override
    public int getItemCount() {
        return categoryList.size();
    }

    public interface OnItemClickListener {
        void onItemClick(int position);
    }

    class CategoryViewHolder extends RecyclerView.ViewHolder {

        private final TextView tvCategoryName;
        private final ImageButton btnCategoryDetail;

        CategoryViewHolder(@NonNull View itemView,
                           final OnItemClickListener listener) {
            super(itemView);

            tvCategoryName = binding.textViewJobName;
            btnCategoryDetail = binding.imageButtonCategoryDetail;

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (listener != null) {
                        int position = getAdapterPosition();
                        if (position != RecyclerView.NO_POSITION) {
                            listener.onItemClick(position);
                        }
                    }
                }
            });

            btnCategoryDetail.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (listener != null) {
                        int position = getAdapterPosition();
                        if (position != RecyclerView.NO_POSITION) {
                            listener.onItemClick(position);
                        }
                    }
                }
            });
        }
    }
}

File: app/src/main/java/com/tt/handsomeman/adapter/CategoryAdapter.kt
package com.tt.handsomeman.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.tt.handsomeman.R
import com.tt.handsomeman.model.Category

class CategoryAdapter(private val context: Context, private val categoryList: List<Category>) : RecyclerView.Adapter<CategoryAdapter.MyViewHolder>() {

    private var mListener: OnItemClickListener? = null
    private val layoutInflater: LayoutInflater

    init {
        layoutInflater = LayoutInflater.from(context)
    }

    interface OnItemClickListener {
        fun onItemClick(position: Int)
    }

    fun setOnItemClickListener(listener: OnItemClickListener) {
        mListener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val item = layoutInflater.inflate(R.layout.item_category, parent, false)
        return MyViewHolder(item, mListener)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val category = categoryList[position]
        holder.tvCategoryName.text = category.name
    }

    override fun getItemCount(): Int {
        return categoryList.size
    }

    inner class MyViewHolder(itemView: View, listener: OnItemClickListener?) : RecyclerView.ViewHolder(itemView) {

        internal var tvCategoryName: TextView
        internal var btnCategoryDetail: ImageButton

        init {

            tvCategoryName = itemView.findViewById(R.id.textViewJobName)
            btnCategoryDetail = itemView.findViewById(R.id.imageButtonCategoryDetail)

            itemView.setOnClickListener {
                if (listener != null) {
                    val position = adapterPosition
                    if (position != RecyclerView.NO_POSITION) {
                        listener.onItemClick(position)
                    }
                }
            }

            btnCategoryDetail.setOnClickListener {
                if (listener != null) {
                    val position = adapterPosition
                    if (position != RecyclerView.NO_POSITION) {
                        listener.onItemClick(position)
                    }
                }
            }
        }
    }
}